test/cli/test_commands.py

```diff
@@ -798,6 +799,39 @@ def test_pre_run_hook_module_not_found(testdir, cli):
     assert lines[9] == "Aborted!"
 
 
+def test_add_case(testdir, cli, schema_url):
+    module = testdir.make_importable_pyfile(
+        hook="""
+            import schemathesis
+            import click
+
+            @schemathesis.hooks.register
+            def add_case(case):
+                if not case.headers:
+                    case.headers = dict()
+                case.headers["copy"] = "this is a copied case"
+                return case
+
+            @schemathesis.register_check
+            def add_case_check(response, case):
+                if case.headers and case.headers.get("copy") == "this is a copied case":
+                    # we will look for this output
+                    click.echo("The case was added!")
+            """
+    )
+
+    result = cli.main(
+        "--pre-run", module.purebasename, "run", "-c", "add_case_check", schema_url, "--hypothesis-max-examples=1"
+    )
+
+    assert result.exit_code == ExitCode.OK
+    # One additional case created for two endpoints - /api/failure and /api/success.
+    assert result.stdout.count("The case was added!") == 2
+    # resetting hooks and checks to ensure other tests start with clean slate
+    unregister_all()
+    reset_checks()
```

Comment:
To ensure that these lines are executed it would be more reliable to put them into a separate fixture:

```
@pytest.fixture
def reset_hooks():
    yield
    unregister_all()
    reset_checks()
```

And then add it as a dependency for this test:

```
@pytest.mark.usefixtures("reset_hooks")
def test_add_case(testdir, cli, schema_url):
    ...
```